trace-normalization/src/normalization/tests/normalize_utils_tests.rs

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present
+// Datadog, Inc.
+
+#[cfg(test)]
+mod normalize_tests {
+
+    use crate::errors;
+    use crate::normalize_utils;
+
+    #[test]
+    fn test_normalize_name() {
+        let test_tuples: [(&str, &str, Option<errors::NormalizeErrors>); 5] = [
```

Comment:
Nit: One advanced way to write table tests like this is to use the duplicate crate
https://crates.io/crates/duplicate
Eg
```
#[duplicate_item(
  test_name input  expected expected_err;
  ["test_normalize_empty_string"] [ "" ]  [ "" ] [None];
)]
#[test]
fn test_name() {
          match normalize_utils::normalize_name(input.to_string()) {
              Ok(val) => {
                  assert_eq!(expected_err, None);
                  assert_eq!(val, expected);
              },
              Err(err) => {
                  assert_eq!(err, expected_err.unwrap());
              }
          }
}
```
It's good because it saves a lot of typing, and you will be individual unit test results.

Reply:
great idea, implemented